From e8c9e30ecdfa5c7519e1543504afdc2440eae99d Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Mon, 18 Dec 2017 17:21:43 +0800
Subject: [PATCH 3120/5242] MLK-17237 staging: typec: return error when
 tcpci_parse_config has failed

commit  c02f40f91e057bf3e6d061764f25329a6af99bf3 from
https://source.codeaurora.org/external/imx/linux-imx.git

Otherwise, the probe would be considered successfully, and without
do any destroy jobs, eg, the interrupt has still registered.

Acked-by: Jun Li <jun.li@nxp.com>
Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/staging/typec/tcpci.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/typec/tcpci.c b/drivers/staging/typec/tcpci.c
index c84dd00..6b543e2 100644
--- a/drivers/staging/typec/tcpci.c
+++ b/drivers/staging/typec/tcpci.c
@@ -668,7 +668,7 @@ static irqreturn_t _tcpci_irq(int irq, void *dev_id)
 static int tcpci_parse_config(struct tcpci *tcpci)
 {
 	struct tcpc_config *tcfg;
-	int ret = 0;
+	int ret = -EINVAL;
 
 	tcpci->controls_vbus = true; /* XXX */
 
@@ -781,8 +781,10 @@ static int tcpci_parse_config(struct tcpci *tcpci)
 		device_property_read_u32(tcpci->dev, "max-snk-mw",
 						&tcfg->max_snk_mw) ||
 		device_property_read_u32(tcpci->dev, "op-snk-mw",
-						&tcfg->operating_snk_mw))
+						&tcfg->operating_snk_mw)) {
+		ret = -EINVAL;
 		goto snk_setting_wrong;
+	}
 
 	return 0;
 
-- 
1.7.9.5

